### ISSUE TITLE
Enhancement: Slider should support "step" change events 

### ISSUE THREAD

**Issue by gavinr** [Esri/calcite-components#3011]
### Description

If using "steps", it would be nice to have an event only when the slider pops into one of the steps (calciteSliderStep event, or something like that)

### Acceptance Criteria 

Can set up an event listener that only gets called when the slider moves into a new "step" position.

### Relevant Info 

Originally requested here: https://github.com/Esri/calcite-components/issues/2652#issuecomment-911931656

#### Which Component

- `calcite-slider`

cc @jcfranco @anveshmekala

**Comment by julio8a** [comment]
There is a task in progress that should address this: https://github.com/Esri/calcite-components/issues/2652

**Comment by anveshmekala** [comment]
@julio8a this issue is an extension for #2652.

**Comment by jcfranco** [comment]
@gavinr I forgot to do this sooner, but can you share an example or provide more info on your use case? When `steps` are configured, the events are only emitted when the thumbs go between steps. Once #2562 lands, you'll have access to both input and change events.

**Comment by gavinr** [comment]
When using this component to control which layer is on the map, if you setup a slider like this, you can see on the event we have to check if it changed. Would like to not have to do that and only get an event once per change.

```
<calcite-slider
        on:calciteSliderChange="{(e) => {
          // Check this and only run if it CHANGED, due to
          // https://github.com/Esri/calcite-components/issues/2652#issuecomment-911931656
          .......
        }}"
        min="1"
        max="{layerTimes_value.length}"
        value="1"
        step="1"
        ticks="1"></calcite-slider>

```

I am not aware of `steps` - only the `step` property. Can you tell me more about `steps` and where I can find that documentation? Maybe that's a solution/workaround.

**Comment by jcfranco** [comment]
My bad, I meant to write `step`. Going back to this issue, #2562 should address this concern. To summarize it will update the slider to:

1. emit `calciteSliderInput` events whenever the value changes
2. emit `calciteSliderChange` events whenever the value changes (except updates between thumb dragging)
3. only emit either of the above when the underlying value has truly changed

**Comment by gavinr** [comment]
Great, yes sounds like #2652 will address the concern. I'll close this for now, and re-open if needed after that issue is completed.
